cmake/FindPackageOrElseAddSubdirectory.cmake

```
@@ -0,0 +1,18 @@
+# Macro for finding a system provided library
+# that fulfills the required minimum version
+# or else including the vendored library
+# from a local git subdirectory
+macro(find_package_or_else_add_subdirectory PACKAGE_NAME MIN_REQUIRED_VERSION SUBDIRECTORY)
+  find_package(${PACKAGE_NAME} ${MIN_REQUIRED_VERSION} QUIET)
+  if (NOT ${${PACKAGE_NAME}_FOUND} OR ("${${PACKAGE_NAME}_VERSION}" VERSION_LESS "${MIN_REQUIRED_VERSION}"))
+    message(STATUS "Using ${PACKAGE_NAME} from submodule: ${SUBDIRECTORY}")
+
+    if(${ARGC} GREATER 3)
+      add_subdirectory(${SUBDIRECTORY} ${ARGN})
+    else()
+      add_subdirectory(${SUBDIRECTORY})
+    endif()
+  else()
+    message(STATUS "Found ${PACKAGE_NAME}: (Found version \"${${PACKAGE_NAME}_VERSION}\") (Required is at least version \"${MIN_REQUIRED_VERSION}\")")
```

```suggestion
    message(STATUS "Found ${PACKAGE_NAME} ${${PACKAGE_NAME}_VERSION} (minimum required version ${MIN_REQUIRED_VERSION})") 
```
It would be useful to also print a path to found package, but it is not critical.

---

The requested changes are done.

The path printing is tricky because for example the current [gtest code](https://github.com/google/googletest/blob/main/googletest/CMakeLists.txt#L116) doesn't even set the `GTEST_INCLUDE_DIRS`, the way I think it should. And `GTEST_INCLUDE_DIRS` or such are useless, empty, so we would have to look for what has been last added to the `INCLUDE_DIRECTORIES`, and get it out of there somehow.